client/src/templates/Challenges/exam/show.tsx

```diff
@@ -615,6 +627,26 @@ class ShowExam extends Component<ShowExamProps, ShowExamState> {
                   {title}
                 </ChallengeTitle>
                 <Spacer size='medium' />
+
+                {qualifiedForExam ? (
+                  <Alert id='qualified-for-exam' bsStyle='info'>
+                    <p>{t('learn.qualified-for-exam')}</p>
+                  </Alert>
+                ) : (
+                  <Alert id='not-qualified-for-exam' bsStyle='danger'>
+                    <p>
+                      {t('learn.not-qualified-for-exam')}{' '}
+                      {t('learn.incomplete-challenges')}
+                    </p>
+                    <Spacer size='small' />
+                    <ul>
+                      {incompletePrequisites.map((challengeTitle, i) => (
+                        <li key={i}>{challengeTitle}</li>
```

Comment:
```suggestion
                      {missingPrequisites.map(({ title, id }) => (
                        <li key={id}>{title}</li>
```
Let's not use the array index - it's a "pitfall" according to [their docs](https://react.dev/learn/rendering-lists#why-does-react-need-keys). It's fine as long as the prerequisite array doesn't change over time, but slightly questionable patterns have a way of spreading.

Also, it's trivial to use the ids here